mars/dataframe/base/transpose.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 1999-2021 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import pandas as pd
+from ..operands import DataFrameOperand, DataFrameOperandMixin
+from ...core import OutputType
+from ..utils import parse_index
+
+
+def reverse(x):
+    if x is None:
+        return
+    return x[::-1]
+
+
+class DataFrameTranspose(DataFrameOperand, DataFrameOperandMixin):
+
+    def __init__(self, **kw):
+        super().__init__(**kw)
+        self.output_types = [OutputType.dataframe]
+
+    def __call__(self, arg):
+        new_shape = arg.shape[::-1]
+        dtypes = arg.dtypes
+        columns_value = arg.index_value
+        index_value = parse_index(dtypes.index, store_data=True)
+        return self.new_dataframe([arg], shape=new_shape, dtypes=dtypes,
```

Comment:
Usually, arg is just a DataFrame or Series, we could infer the `dtypes`, `index_value`, `columns_value`, `shape` by the input DataFrame. For `shape` as example, `new_shape=arg[0].shape[::1]`.

Reply:
In addition, we don't recommend `store_data=True` for index_value as index may a huge series.